Sming/SmingCore/Platform/System.h

```diff
@@ -85,43 +93,57 @@ class SystemClass
 	}
 
 	/** @brief System initialisation
+	 *  @retval bool true on success
+	 *  @note Called by user_main: applications should not call this function or the task queue
+	 *  will be re-initialised and any currently queued tasks won't be called.
 	 */
-	void initialize();
+	static bool initialize();
```

Comment:
I was looking at the class changes and... having a class that has a lot of static methods in my humble experience always spells problems. One of the solutions to these problems is having a singleton. Although in our cause I don't think we need this. We can still have our global `System` object which is instance of the `SystemClass` and use that anywhere in our code (apart from hardcoding the instance inside of the class itself) 

Why not make the following instead:
- make the following methods non-static initialize, isReady, restart, setCpuFrequency, getCpuFrequency, deepSleep, onReady 
- make `SystemState state`  static. This way the queue, hopefully, can be initialized only once (after adding back
```
if(state != eSS_None)	
		return;	
	state = eSS_Intializing;
``` 
in the initialization method.
- find all places in the code where `SystemClass` was used in a static context and replace it with `System` instance.

Reply:
Once you make the suggested changes I will take care to merge this PR.

Reply:
OK. I posted (twice) on the Espressif forum regarding the `system_init_done_cb` vs task-queue thing, but it's not even been approved by their moderator yet so that's a bust I think. Do you know of any other way to elicit a response from them?

Reply:
> Do you know of any other way to elicit a response from them?

I know two guys from Espressif. Will have to ask them directly. Can you paste here the post message for reference?

Reply:
Subject: `NON-OS SDK Task Queue vs. system_init_done_cb()`
Content: `I have multiple separate items to initialise in my [code]system_init_done_cb[/code] callback. Is it OK to use the task queue for this instead?`

Reply:
> Once you make the suggested changes I will take care to merge this PR.

OK, have submitted changes. I have,however, left `initialise()` as static because it's called from `user_main.cpp` _before_ `cpp_core_initialize()`.